File: gdscript/src/main/java/gdscript/psi/GdParam.java
package gdscript.psi;

import com.intellij.psi.PsiElement;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public interface GdParam extends PsiElement {

  @Nullable
  GdAssignTyped getAssignTyped();

  @Nullable
  GdExpr getExpr();

  @Nullable
  GdTyped getTyped();

  @NotNull
  GdVarNmi getVarNmi();

  @NotNull
  String getReturnType();

}

File: gdscript/src/main/java/gdscript/psi/GdParam.kt
package gdscript.psi

import com.intellij.psi.PsiElement

interface GdParam : PsiElement {
    val assignTyped: GdAssignTyped?

    val expr: GdExpr?

    val typed: GdTyped?

    val varNmi: GdVarNmi

    val returnType: String
}
